src/dsp_tools/utils/excel_to_json/excel_to_json_properties.py

```diff
@@ -0,0 +1,470 @@
+    return True
+
+
+def _search_convert_numbers(value_str: str) -> str | int | float:
+    """
+    This function takes a string and searches if the string contains a float or an integer. In those cases, it converts
+    the string to the corresponding data type. If it is not a float or integer, it returns the string as is.
+
+    Args:
+        value_str: The value which is checked and may be converted
+
+    Returns:
+        A int if the string was an integer, float if the string was a float or str if it was neither
+    """
+    if regex.search(r"^\d+\.\d+$", value_str):
+        return float(value_str)
+    elif regex.search(r"^\d+$", value_str):
+        return int(value_str)
+    else:
+        return value_str
+
+
+def _unpack_gui_attributes(gui_str: str) -> dict[str, str]:
+    """
+    This function takes a string which contains the gui_attributes if they are formatted according to the documentation
+    a correct dictionary with the attribute name and its value is created. If the string is not formatted correctly,
+    then the resulting list may not be in the expected format, this raises an IndexError. Only very broad errors
+    can be caught with this function. Errors regarding the content will be diagnosed when the json is validated.
+
+    Args:
+        gui_str: A string containing the gui_attributes
+
+    Returns:
+        A dictionary with the gui_attribute name as key and the attribute as value.
+
+    Raises:
+        IndexError: if the sub-lists do not contain each two items
+    """
+    # Create a list with several attributes
+    gui_list = [x.strip() for x in gui_str.split(",") if not x.strip() == ""]
+    # create a sub list with the kex value pair of the attribute if it is an empty string we exclude it.
+    # this error will be detected when checking for the length of the lists
+    sub_gui_list = [[sub.strip() for sub in x.split(":") if sub.strip() != ""] for x in gui_list]
+    # if not all sublist contain two items, something is wrong with the attribute
+    if not all(len(sub) == 2 for sub in sub_gui_list):
+        raise IndexError
+    return {sub[0]: sub[1] for sub in sub_gui_list}
+
+
+def _format_gui_attribute(attribute_str: str) -> dict[str, str | int | float]:
+    """
+    This function takes a string containing the information about the gui_attributes. First, a dictionary is created
+    separating different attribute names and their values; in the second step it converts the numbers which are stored
+    as strings into int or float data types.
+
+    Args:
+        attribute_str: A string containing the attributes
+
+    Returns:
+        A dictionary with the attribute name as a key and the attribute as value.
+
+    Raises:
+        IndexError if the attributes are not formatted correctly
+    """
+    attribute_dict = _unpack_gui_attributes(gui_str=attribute_str)
+    return {attrib: _search_convert_numbers(value_str=val) for attrib, val in attribute_dict.items()}
```

Comment:
I like this styli with ultra-short functions and speaking names; but if code is written like that, such an extent of documentation is somewhat redundant and actually makes the code harder to follow

Reply:
Thanks! I have shortened it some.